sound/soc/sof/intel/hda-ipc.c

```diff
@@ -237,14 +237,12 @@ irqreturn_t hda_dsp_ipc_irq_thread(int irq, void *context)
 	return IRQ_HANDLED;
 }
 
-/* is this IRQ for ADSP ? - we only care about IPC here */
-irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
+static bool is_ipc_irq(struct snd_sof_dev *sdev)
 {
-	struct snd_sof_dev *sdev = context;
-	int ret = IRQ_NONE;
+	int ret = false;
 	u32 irq_status;
 
-	spin_lock(&sdev->hw_lock);
+	spin_lock_irq(&sdev->hw_lock);
```

Comment:
why do we need a change here? is this is because we use this routine both from handler and interrupt threads, than you need to add a comment.

Reply:
Yes, is_ipc_irq() will be called on interrupt thread and handler, and we don't want another interrupt to lock it while it is locked.